src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.BoxedTypes.cs

```diff
@@ -0,0 +1,396 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
```

Comment:
Somewhere in here we need a giant comment describing what's happening here. I understand what you're doing, but it wouldn't  be obvious from the pile of code without knowing what the PR is supposed to do.